rcl/include/rcl/client.h

```diff
@@ -409,6 +410,37 @@ RCL_PUBLIC
 bool
 rcl_client_is_valid(const rcl_client_t * client);
 
+/// Set the on new response callback function for the client.
+/**
+ * This API sets the callback function to be called whenever the
+ * client is notified about a new response.
+ *
+ * \sa rmw_client_set_on_new_response_callback for details about this function.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | Yes
+ * Uses Atomics       | Maybe [1]
+ * Lock-Free          | Maybe [1]
+ * <i>[1] rmw implementation defined</i>
+ *
+ * \param[in] client The client on which to set the callback
+ * \param[in] callback The callback to be called when new responses arrive
+ * \param[in] user_data Given to the callback when called later, may be NULL
+ * \return `RCL_RET_OK` if callback was set to the listener, or
+ * \return `RCL_RET_INVALID_ARGUMENT` if `client` or `callback` is NULL, or
```

Comment:
to unset the callback, application will call this function with `callback == null` ? or is there any other way to unset the callback? same questions goes to other interfaces.